Fixes several errors in remove_handles

hdlim15 opened the pull request:
I updated the regular expression from remove_handles so that it now achieves the following:
 - Handles can begin with numbers
 - Handles are allowed to follow any of the following characters: `~()-=+\|[]{};:'\"/?.,<> \n
 - Handles are allowed to precede the following characters: !#$%&*
 - Handles have a max length of 20 (15 today, but it was 20 in the past so the max length handle you can tweet is 20)
 - Two @ in a string with only alphanumeric characters and _ between them are not handles
 - Unless the first handle is 20 characters long and there is at least one more character before the next @

The above changes may seem oddly specific, and that's because they are. These changes were not found through any Twitter documentation, but rather tweeting various test cases and observing which were tagged as handles by Twitter.

The following is a link to a gist that tests the previous regex against my suggested change:
(HINT: mine passes all the tests and the previous regex fails nearly all of them)
https://gist.github.com/hdlim15/2419dd8f6bb3987f4a3536a4008b0544

Commit:
Fixes several errors in remove_handles

I updated the regular expression from remove_handles so that it now achieves the following:
 - Handles can begin with numbers
 - Handles are allowed to follow any of the following characters: `~()-=+\|[]{};:'\"/?.,<> \n
 - Handles are allowed to precede the following characters: !#$%&*
 - Handles have a max length of 20 (15 today, but it was 20 in the past so the max length handle you can tweet is 20)
 - Two @ in a string with only alphanumeric characters and _ between them are not handles
 - Unless the first handle is 20 characters long and there is at least one more character before the next @

The above changes may seem oddly specific, and that's because they are. These changes were not found through any Twitter documentation, but rather tweeting various test cases and observing which were tagged as handles by Twitter.

The following is a link to a gist that tests the previous regex against my suggested change:
(HINT: mine passes all the tests and the previous regex fails nearly all of them)
https://gist.github.com/hdlim15/2419dd8f6bb3987f4a3536a4008b0544

stevenbird commented:
@hdlim15: thanks for this, especially for including test code. Would you mind adding tests to `nltk/test` or `nltk/test/unit` please, so we ensure future contributions don't introduce any regressions?

Commit:
Small improvement for consistency

Replace handles with a space (i.e. " ") instead of the empty string (i.e. "") to ensure that text on either side of removed handles are tokenized correctly

hdlim15 commented:
Hello,

I just submitted a pull request in test/unit. Just wondering, does this mean I can get added to the list of contributors?

Hyuckin David Lim

> On Nov 23, 2016, at 5:16 AM, Steven Bird notifications@github.com wrote:
> 
> @hdlim15 https://github.com/hdlim15: thanks for this, especially for including test code. Would you mind adding tests to nltk/test or nltk/test/unit please, so we ensure future contributions don't introduce any regressions?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub https://github.com/nltk/nltk/pull/1521#issuecomment-262476182, or mute the thread https://github.com/notifications/unsubscribe-auth/AO4U_pRTqa6gtuNyUE1tCdXJgR0l81lSks5rBBJmgaJpZM4K6RLe.

stevenbird commented:
@hdlim15 yes please add yourself to our list of contributors (via an extra commit to your `hdlim15:patch-1` branch)

Commit:
Update AUTHORS.md

stevenbird commented:
Thanks @hdlim15!
